target/linux/ipq40xx/files/arch/arm/boot/dts/qcom-ipq4019-whw03v2.dts

```diff
@@ -0,0 +1,528 @@
+
+    aliases {
+        led-boot = &led_blue;
+        led-failsafe = &led_red;
+        led-running = &led_green;
+        led-upgrade = &led_red;
+    };
+
+    // The arguments rootfstype and ro are needed
+    // to override the default bootargs
+    chosen {
+        bootargs-append = " root=/dev/ubiblock0_0 rootfstype=squashfs ro";
+        stdout-path = &blsp1_uart1;
+    };
+
+    soc {
+        ess_tcsr@1953000 {
+            compatible = "qcom,tcsr";
+            reg = <0x1953000 0x1000>;
+            qcom,ess-interface-select = <TCSR_ESS_PSGMII>;
+        };
+
+
+        tcsr@1949000 {
+            compatible = "qcom,tcsr";
+            reg = <0x1949000 0x100>;
+            qcom,wifi_glb_cfg = <TCSR_WIFI_GLB_CFG>;
+        };
+
+        tcsr@194b000 {
+            compatible = "qcom,tcsr";
+            reg = <0x194b000 0x100>;
+            qcom,usb-hsphy-mode-select = <TCSR_USB_HSPHY_HOST_MODE>;
+        };
+
+        tcsr@1957000 {
+            compatible = "qcom,tcsr";
+            reg = <0x1957000 0x100>;
+            qcom,wifi_noc_memtype_m0_m2 = <TCSR_WIFI_NOC_MEMTYPE_M0_M2>;
+        };
+    };
+
+
+    keys {
+        compatible = "gpio-keys";
+
+        reset {
+            label = "reset";
+            gpios = <&tlmm 18 GPIO_ACTIVE_LOW>;
+            linux,code = <KEY_RESTART>;
+        };
+    };
+};
+
+
+&tlmm {
+    mdio_pins: mdio_pinmux {
+        mux_1 {
+            pins = "gpio6";
+            function = "mdio";  
+            bias-pull-up;
+        };
+
+        mux_2 {
+            pins = "gpio7";
+            function = "mdc";
+            bias-pull-up;
+        };
+    };
+
+    i2c_0_pins: i2c_0_pinmux {
+        mux {
+            function = "blsp_i2c0";
+            pins = "gpio20", "gpio21";
+            bias-disable;
+        };
+    };
+
+    serial_0_pins: serial0_pinmux {
+        mux {
+            pins = "gpio16", "gpio17";
+            function = "blsp_uart0";
+            bias-disable;
+        };
+    };
+
+    serial_1_pins: serial1_pinmux {
+        mux {
+            pins = "gpio8", "gpio9", "gpio10", "gpio11";
+            function = "blsp_uart1";
+            bias-disable;
+        };
+    };
+
+    spi_0_pins: spi_0_pinmux {
+        mux {
+            function = "blsp_spi0";
+            pins = "gpio13", "gpio14", "gpio15";
+            drive-strength = <12>;
+            bias-disable;
+        };
+
+        mux_cs {
+            pins = "gpio12";
+            drive-strength = <2>;
+            bias-disable;
+            output-high;
+        };
+    };
+    
+    spi_1_pins: spi_1_pinmux {
+        mux_1 {
+            function = "blsp_spi1";
+            pins = "gpio44", "gpio46","gpio47";
+            bias-disable;
+        };
+
+        mux_2 {
+            pins = "gpio31", "gpio45", "gpio49";
+            function = "gpio";
+            bias-pull-up;
+            output-high;
+        };
+
+        mux_3 {
+            pins = "gpio42";
+            function = "gpio";
+            input;
```

Comment:
This property is new to me and I dont find the bindings for it.
Can you point me to the upstream support for it?

It should work just fine without it

Reply:
I took this pin from the vendor firmware's devicetree:
```
   spi_1_pinmux {
    linux,phandle = <0x4b>;
    phandle = <0x4b>;

    cs {
     bias-pull-up;
     function = "gpio";
     pins = "gpio45";
    };

    host_int {
     function = "gpio";
     input;
     pins = "gpio42";
    };

    mux {
     bias-disable;
     function = "blsp_spi1";
     pins = "gpio44\0gpio46\0gpio47";
    };

    reset {
     function = "gpio";
     output-high;
     pins = "gpio49";
    };

    wake {
     function = "gpio";
     output-high;
     pins = "gpio31";
    };
   };
 ```
 
 It was originally called `host_int` and it seems to be used to handle interrupts from the spidev device (Zigbee co-processor). 
 I am still working on the zigbee integration, I need to get access to the stack SDK and for that I need to get a dev board, but I would leave this pin as it is used by the chip.
 I can rename it to `host_int` to make its purpose more intelligible.

Reply:
Yeah, please make the names verbose.

Reply:
done